tests/unit/s2n_ktls_test_utils_test.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "testlib/s2n_ktls_test_utils.h"
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+
+#define S2N_TEST_RECORD_TYPE 43
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* test data */
+    uint8_t test_data[S2N_TLS_MAXIMUM_FRAGMENT_LENGTH] = { 0 };
+    struct s2n_blob test_data_blob = { 0 };
+    EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
+    EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
+
+    /* Test the mock IO stuffer implementation can send/recv records */
+    {
+        /* Test sending send/recv 0 bytes */
+        {
+            DEFER_CLEANUP(struct s2n_connection *server = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            DEFER_CLEANUP(struct s2n_connection *client = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            DEFER_CLEANUP(struct s2n_test_ktls_io_pair io_pair = { 0 },
+                    s2n_ktls_io_pair_free);
+            EXPECT_OK(s2n_test_init_ktls_stuffer_io(server, client, &io_pair));
+
+            size_t to_send = 0;
+            /* Init send msghdr */
+            struct iovec send_msg_iov = { 0 };
+            send_msg_iov.iov_base = test_data;
+            send_msg_iov.iov_len = to_send;
+            struct msghdr send_msg = { 0 };
+            send_msg.msg_iov = &send_msg_iov;
+            send_msg.msg_iovlen = 1;
+            /* sendmsg */
+            ssize_t bytes_written = s2n_test_ktls_sendmsg_stuffer_io(server, &send_msg, S2N_TEST_RECORD_TYPE);
+            EXPECT_EQUAL(bytes_written, to_send);
+
+            EXPECT_EQUAL(io_pair.client_in.invoked_count, 1);
```

Comment:
What exactly is this invoked_count testing? I wonder if there's a better name for this variable so I automatically recognize it.